src/iexamine.cpp

```diff
@@ -855,6 +855,26 @@ void iexamine::toilet( player &p, const tripoint &examp )
     }
 }
 
+/** Toggle the lights in a overmap terrain*/
+void iexamine::toggle_lights( player &/*p*/, const tripoint &examp )
+{
+    map &here = get_map();
+    std:: string flag = "L_OFF";
+    std::string message = g->m.furn( examp ).obj().message;
+    std::string prompt = g->m.furn( examp ).obj().prompt;
+
+    // Checks to turn on or off
+    if( !here.has_flag_furn( flag, examp ) ) {
+        flag = "L_ON";
+    };
```

Comment:
```suggestion
    const auto flag = here.has_flag_furn( flag, examp ) ? "L_OFF" : "L_ON";
```
avoid mutation where possible because it makes code harder to read

Reply:
This line gives me the error `a variable declared with an auto type specifier cannot appear in its own initializer	` in Visual Studio.

Reply:
my mistake, it should have been `"L_OFF"`. also we can see why mutation is not often desired: it's easy to make little mistakes like this.

Reply:
Alright thank you